server/app/views/questiontypes/ApplicantQuestionRendererImpl.java

```diff
@@ -23,10 +27,18 @@
  */
 abstract class ApplicantQuestionRendererImpl implements ApplicantQuestionRenderer {
 
+  /** Whether the question is comprised of a single input field or multiple. */
+  protected enum InputFieldType {
+    SINGLE,
+    COMPOSITE
+  }
+
   protected final ApplicantQuestion question;
+  private final InputFieldType inputFieldType;
 
-  ApplicantQuestionRendererImpl(ApplicantQuestion question) {
+  ApplicantQuestionRendererImpl(ApplicantQuestion question, InputFieldType inputFieldType) {
     this.question = checkNotNull(question);
+    this.inputFieldType = inputFieldType;
```

Comment:
I believe enums can be null in Java. Can you add a checkNotNull(inputFieldType) here for consistency with other constructors?

Reply:
Ah thanks, I didn't realize that. Done!